src/Analyser/Scope.php

```diff
@@ -1113,7 +1087,15 @@ private function resolveType(Expr $node): Type
 			}
 			if ($constantClassType->hasConstant($constantName)) {
 				$constant = $constantClassType->getConstant($constantName);
-				return $this->getTypeFromValue($constant->getValue());
+				$constantType = $this->getTypeFromValue($constant->getValue());
+				if (
+					$constantClassType instanceof ObjectType &&
```

Comment:
I'd rather have `TypeUtils::getDirectClassNames($constantClassType)` and checking that `count($directClassNames) === 1` and use `$directClassNames[0]`.

Reply:
This part can be removed if we want to support only global constants.